src/chains/base.ts

```diff
@@ -8,4 +8,38 @@ export const base = {
     ...viemBase.contracts,
     ...opStackL2ContractAddresses,
   },
-} satisfies Chain
+  optimismConfig: {
+    l1Contracts: {
+      optimismL1CrossDomainMessenger: {
+        address: '0x866E82a600A1414e583f7F13623F1aC5d58b0Afa',
+        blockCreated: 17482143,
+      },
+      optimismL1Erc721Bridge: {
+        address: '0x608d94945A64503E642E6370Ec598e519a2C1E53',
+        blockCreated: 17482143,
+      },
+      optimismL1StandardBridge: {
+        address: '0x3154Cf16ccdb4C6d922629664174b904d80F2C35',
+        blockCreated: 17482143,
+      },
+      optimismL2OutputOracle: {
+        address: '0x56315b90c40730925ec5485cf004d835058518A0',
+        blockCreated: 17482143,
+      },
+      optimismPortal: {
+        address: '0x49048044D57e1C92A77f79988d21Fa8fAF74E97e',
+        blockCreated: 17482143,
+      },
+    },
+  },
+} satisfies OpStackChain
+
+export type OpStackChain = Chain & {
```

Comment:
Alternative is to export only the config not the chain

`export const baseConfig: OptimismConfig`

But having it literally be a chain is nice

Reply:
Is there a specific reason @jxom we can't get the optimism config on the wagmi/viem object? Getting it to just natively be on the wagmi/viem chain object instead of being it's own seperate type is much preferred

Reply:
@roninjin10 I think we can, I believe the concern was more on overloading things like `mainnet` with contracts that not everyone uses. But as you realized from the start, having these live on the L2 Chain makes more sense :)

Reply:
Nice I do think the l1 property is more general than just OP chains and useful for all rollups. I mentioned before but being able to do chain.l1 instead of `chain.id === 10 ? 1 : chain.id === 420 ? 5 ....` eliminated an entire class of bugs we were constantly fighting against and deleted so much code